### ISSUE TITLE
Error rendering hierarchical glossary terms with same name

### ISSUE THREAD

**Issue by cutz** [open-metadata/OpenMetadata#12342]
**Affected module**
I believe this is purely a UI display issue as the data model fetched via the API looks correct.

**Describe the bug**

When creating a glossary of nested terms, certain structures with overlapping term names, result in the glossary tree being rendered incorrectly.

**To Reproduce**

Create a test glossary with 5 nested terms as follows.

Test Glossary (glossary)

* Level 1
  * Level 2
    * Level 3
      * Level 4
        * Level 2 _(This is nested beneath Level 4)_

In the UI Level 4's child is rendered as a child of Level 1.

![Screenshot 2023-07-10 at 2 48 23 PM](https://github.com/open-metadata/OpenMetadata/assets/2072268/fc2e2214-e342-4e20-9013-b188d6c4bb78)

Here is a condensed snippet of json representing the parents, names, and fully qualified names of the glossary terms at play.

```json
{
  "parent": null,
  "name": "Level 1",
  "fullyQualifiedName": "Test Glossary.Level 1",
  "description": "Level 1",
  "glossary": "Test Glossary"
}
{
  "parent": "Test Glossary.Level 1",
  "name": "Level 2",
  "fullyQualifiedName": "Test Glossary.Level 1.Level 2",
  "description": "Level 2",
  "glossary": "Test Glossary"
}
{
  "parent": "Test Glossary.Level 1.Level 2.Level 3.Level 4",
  "name": "Level 2",
  "fullyQualifiedName": "Test Glossary.Level 1.Level 2.Level 3.Level 4.Level 2",
  "description": "This is actually Level 5",
  "glossary": "Test Glossary"
}
{
  "parent": "Test Glossary.Level 1.Level 2",
  "name": "Level 3",
  "fullyQualifiedName": "Test Glossary.Level 1.Level 2.Level 3",
  "description": "Level 3",
  "glossary": "Test Glossary"
}
{
  "parent": "Test Glossary.Level 1.Level 2.Level 3",
  "name": "Level 4",
  "fullyQualifiedName": "Test Glossary.Level 1.Level 2.Level 3.Level 4",
  "description": "Level 4",
  "glossary": "Test Glossary"
}

```

**Expected behavior**

I would expect the tree of glossary Terms to render as 

* Level 1
  * Level 2
    * Level 3
      * Level 4
        * Level 2 _(This is nested beneath Level 4)_

when in fact it is rendering as

* Level 1
  * Level 2
    * Level 3
      * Level 4
  * Level 2 _(This is nested beneath Level 4)_

**Version:**

OpenMetadata Server version 1.1.0

**Comment by Ashish8689** [comment]
@cutz This is a duplicate issue and it is already fixed in this PR https://github.com/open-metadata/OpenMetadata/issues/12425 and will be available in `1.1.1 release`

**Comment by Ashish8689** [comment]
@cutz This is a duplicate issue and it is already fixed in this PR https://github.com/open-metadata/OpenMetadata/issues/12425 and will be available in 1.1.1 release
